File: bisection_method.nim
import math
const acc = 1.0e-7

proc f(x: float): float =
    return ln(x) + x - 3

proc bisection(a1, b1: float) =
    var
        a = a1
        b = b1
    if f(a1)*f(b1) >= 0:
        echo "You did't assumed right the root is between a and b\n"
        return
    var c: float
    while (b-a) >= acc:
        c = (a+b)/2
        if f(c) == 0:
            break
        elif f(a)*f(c) < 0:
            b = c
        else:
            a = c
    echo "the value of root is ", c;

bisection(0.0, 5.0)

File: bisection_method.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
#define EPSILON 1.0e-7
using namespace std;

double f(double x)
{
    return log(x) + x - 3;
}

void bisection(double a, double b)
{
    if (f(a) * f(b) >= 0)
    {
        cout << "You did't assumed right the root is between a and b\n";
        return;
    }
    double c;
    while ((b - a) >= EPSILON)
    {
        c = (a + b) / 2;
        if (f(c) == 0)
        {
            break;
        }
        else if (f(c) * f(a) < 0)
        {
            b = c;
        }
        else
        {
            a = c;
        }
    }
    cout << "the value of root is " << setprecision(20) << c;
}

int main()
{
    bisection(0, 5);
    return 0;
}